backend/inspirehep/records/tasks.py

```diff
@@ -96,10 +98,15 @@ def update_references_pointing_to_merged_record(
             referenced_records_in_path = flatten_list(
                 get_value(matched_inspire_record, path[: -len(".$ref")], [])
             )
+
             for referenced_record in referenced_records_in_path:
-                update_reference_if_reference_uri_matches(
-                    referenced_record, merged_record_uri, new_record_uri
-                )
+                update_reference_if_reference_uri_matches(referenced_record, merged_record_uri, new_record_uri)
+            
+            referenced_records_in_path = dedupe_list_of_dicts(referenced_records_in_path)
+            updated_matched_inspire_record = remove_duplicate_refs_from_record(matched_inspire_record, referenced_records_in_path, path)
+            if updated_matched_inspire_record:
+                matched_inspire_record = updated_matched_inspire_record
```

Comment:
I don't get why you can't simply assign the deduped list to the correct path and update a record with it?

Reply:
How would you assign/update it? I had a lot of trouble using the path since it's a string separated with . and if I simply removed the duplicates from the list it doesn't remove them from the ```matched_inspire_record```.
That's why I found the ```set_value_by_path``` function so I could update the affiliations once I have removed the duplicates from "the old version" of the record.